### PR TITLE
peercoin: init 0.6.2

### PR DESCRIPTION
**Pull request by spinus**
###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/altcoins/peercoin.nix [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                  miniupnpc utillinux protobuf libevent ]
+                  ++ optionals withGui [ qt4 qrencode ];
+
+  configureFlags = [ "--with-boost-libdir=${boost.out}/lib" ]
+                     ++ optionals withGui [ "--with-gui=qt4" ];
+
+  buildPhase = ''
+    qmake
+    make
+    find
```
Comment (Mic92) [inline pkgs/applications/altcoins/peercoin.nix]: should be probably removed

Comment (spinus) [inline pkgs/applications/altcoins/peercoin.nix]: thx :-)

**Comment by Mic92** [comment]
peercoind fails to build:

```
these derivations will be built:
  /nix/store/zslm1jfbnrii9yxnpvspjr5518pxdm36-peercoind-0.5.4ppc.drv
building path(s) ‘/nix/store/jwncf41m24nf57mmk1sxqlprvcjdavc0-peercoind-0.5.4ppc’
unpacking sources
unpacking source archive /nix/store/wr1rhy83h8lvv7l1p9jg0v973dfy989w-peercoin-v0.5.4ppc-src
source root is peercoin-v0.5.4ppc-src
patching sources
configuring
no configure script, doing nothing
building
/nix/store/jdpwa3nmjwbcw05hxm89rsgmd6n7bk2s-stdenv/setup: line 917: qmake: command not found
builder for ‘/nix/store/zslm1jfbnrii9yxnpvspjr5518pxdm36-peercoind-0.5.4ppc.drv’ failed with exit code 127
error: build of ‘/nix/store/zslm1jfbnrii9yxnpvspjr5518pxdm36-peercoind-0.5.4ppc.drv’ failed
/run/current-system/sw/bin/nix-shell: failed to build all dependencies
```

**Comment by spinus** [comment]
@Mic92 good catch, I checked only peercoin-qt, I'll try to fix...

**Comment by matthewbauer** [comment]
@GrahamcOfBorg build peercoin

**Comment by matthewbauer** [comment]
@GrahamcOfBorg build altcoins.peercoin altcoins.peercoind

**Review by matthewbauer — CHANGES_REQUESTED** [review]
Checksum looks incorrect

**Comment by spinus** [comment]
Bumped to latest version and updated checksum. Was not sure how to build non gui target so removed that.

**Comment by Mic92** [comment]
@GrahamcOfBorg build altcoins.peercoin

**Review by matthewbauer — APPROVED** [review]

File: pkgs/applications/altcoins/peercoin.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
+{ stdenv, fetchFromGitHub
+, pkgconfig, autoreconfHook
+, openssl, db48, boost, zlib, miniupnpc
+, protobuf, utillinux, qt4, qrencode
```
Comment (matthewbauer) [inline pkgs/applications/altcoins/peercoin.nix]: I think we should use qt5 if possible. Qt4 is soon to be deprecated.

**Review by matthewbauer — CHANGES_REQUESTED** [review]

File: pkgs/applications/altcoins/peercoin.nix (hunk 2 above)
Comment (spinus) [reply pkgs/applications/altcoins/peercoin.nix]: let me check if that even compiles

**Comment by Ekleog** [comment]
(triage) @spinus Are you still planning to come back to this PR?

**Comment by aanderse** [comment]
> (triage) @spinus Are you still planning to come back to this PR?

@spinus ping

**Comment by spinus** [comment]
hello, thanks for ping. Sure, I'll jump on that soon.

File: pkgs/applications/altcoins/peercoin.nix (hunk 2 above)
Comment (aanderse) [reply pkgs/applications/altcoins/peercoin.nix]: Seems like most recent release supports qt5 from the .pro file

**Review by aanderse — CHANGES_REQUESTED** [review]
Let me know if you have any questions.

**Comment by wamserma** [comment]
@spinus Seems this is not really missing in NixOS. Maybe close this PR and provide package via [NUR](https://github.com/nix-community/NUR).

**Comment by spinus** [comment]
@wamserma not bad option. I thought I'll add this here as the arch for adding more tools in that category is easy and prepared. But I guess you could be right :-)

**Pull request closed without merging**